modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx

```diff
@@ -621,3 +663,346 @@ we've made a few updates to the component in v5:
+  Popup Content
+</Popup>
+
+// after
+<Popup.Card with={width}>
+  <Popup.CloseIcon aria-label="Close" onClick={onClose} />
+  <Popup.Heading>Popup Heading</Popup.Heading>
+  <Popup.Body>Popup Content</Popup.Body>
+</Popup.Card>
+```
+
+`Popup.Card` is using `Card`, which is now using `Box`. This means changes to some props:
+
+| Before                         | After                                      |
+| ------------------------------ | ------------------------------------------ |
+| `padding={Popup.Padding.zero}` | `padding="zero"` or `padding={space.zero}` |
+| `depth={depth[0]}`             | `depth={0}`                                |
+| `popupRef={ref}`               | `ref={ref}`                                |
+
+#### Transitioning
+
+We noticed Popups were used in 2 different ways: always rendering and conditional rendering.
+
+```tsx
+// always rendering
+const MyPopup = () => {
+  const targetRef = React.useRef(null)
+  const {stackRef, popperProps, targetProps, closePopup} = usePopup()
+
+  const handleClose = () => {
+    closePopup()
+    targetRef.current.focus() // focus back on target
+  }
+
+  useCloseOnEscape(stackRef, handleClose)
+
+  return (
+    <>
+      <button ref={targetRef} {...targetProps}>Open</button>
+      <Popper {...popperProps}>
+        <Popup>
+          {/* content */}
+          <button onClick={handleClose}>Close</button>
+        </Popup>
+      </Popper>
+    </>
+  )
+}
+
+// conditional rendering
+const MyOpenPopup = ({onClose, targetRef}) => {
+  const {popperProps, closePopup} = usePopup()
+
+  const handleClose = () => {
+    onClose()
+    closePopup()
+    targetRef.current.focus() // focus back on target
+  }
+
+  useCloseOnEscape(stackRef, handleClose)
+
+  return (
+    <Popper {...popperProps}>
+      <Popup>
+        {/* content */}
+        <button onClick={handleClose}>
+      </Popup>
+    </Popper>
+  )
+}
+
+const MyPopup = () => {
+  const targetRef = React.useRef(null)
+  const [open, setOpen] = React.useState(false)
+
+  const onClose = () => {
+    setOpen(false)
+  }
+
+  return (
+    <>
+      <button ref={targetRef} onClick={() => { setOpen(true) }}>
+      {open && <MyOpenPopup onClose={onClose} />}
+    </>
+  )
+}
+```
+
+The difference between the two is subtle, but when always rendering example, the `usePopup` hook
+runs on every render. In the conditional rendering example, the `usePopup` hook only runs when
+`MyPopup` renders it. This means hooks like `useCloseOnEscape` need to function properly in both
+cases, but `open` is not passed to the hook. This caused subtle bugs, and now `useCloseOnEscape` is
+passed a `PopupModel` which does have access to the popup's visible state. `useCloseOnEscape` will
+now only run when the popup is visible, but this means the conditional rendering example will have
+to do extra work because the `target` is out of scope of the `MyOpenPopup` component. The following
+is equivalent to the example in v5:
```

Comment:
```suggestion
The difference between the two is subtle, but in the always rendering example, the `usePopup` hook
runs on every render. In the conditional rendering example, the `usePopup` hook only runs when
`MyPopup` renders it. This means hooks like `useCloseOnEscape` need to function properly in both
cases, but `open` is not passed to the hook. This caused subtle bugs. Now, `useCloseOnEscape` is
passed a `PopupModel` which has access to the popup's visible state. `useCloseOnEscape` will now
only run when the popup is visible, but this means the conditional rendering example will have to do
extra work because the `target` is out of scope of the `MyOpenPopup` component. The following is
equivalent to the example in v5:
```